src/common/log/progressLogs.ts

```diff
@@ -32,6 +35,29 @@ export class ConsoleProgressLog implements ProgressLog {
     public close() { return Promise.resolve(); }
 }
 
+
+export class S3ProgressLog implements ProgressLog {
```

Comment:
I'd put this in its own file. And perhaps we should split out the others also. The grouping doesn't really make sense any more.

Reply:
Actually I'l merge it and the do this